src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/convert_matmul_to_fc.cpp

```diff
@@ -133,25 +135,28 @@ ov::intel_cpu::ConvertMatMulToFC::ConvertMatMulToFC() {
         // Transferring from MatMul representation: [B, I, K] * [B, K, O] = [B, I, O]
         // to FullyConnected representation: [I, K] * [K, O] = [I, O]
 
-        // Weights normalization
-        if (!matmul->get_transpose_b()) {
-            fc_input_b = create_transpose(fc_input_b, matmul->get_friendly_name() + "/transpose_b");
-        }
-
         if (rank_b != 2) {
             ngraph::Dimension K = *(shape_b_aligned.rbegin() + 1);
             NGRAPH_CHECK(K.is_static());
-            std::vector<int64_t> reshape_shape_values = { -1ll, static_cast<int64_t>(K.get_length()) };
-            auto reshape_shape = ngraph::opset1::Constant::create(ngraph::element::i64, ngraph::Shape{ 2 }, reshape_shape_values);
+            std::vector<int64_t> reshape_shape_values = {K.get_length()};
+            const auto& insert_pos = matmul->get_transpose_b() ? reshape_shape_values.begin() : reshape_shape_values.begin() + 1;
+            reshape_shape_values.insert(insert_pos, -1);
+
+            auto reshape_shape = ngraph::opset1::Constant::create(ngraph::element::i32, ngraph::Shape{ 2 }, reshape_shape_values);
             fc_input_b = ov::op::util::make_try_fold<ngraph::opset1::Reshape>(fc_input_b, reshape_shape, false);
             if (!std::dynamic_pointer_cast<ngraph::opset1::Constant>(fc_input_b.get_node_shared_ptr())) {
                 new_ops.push_back(reshape_shape);
             }
             new_ops.push_back(fc_input_b.get_node_shared_ptr());
         }
 
+        // Weights normalization
+        if (!matmul->get_transpose_b()) {
```

Comment:
is "!" supposed to be here?
or it should be
if(matmul->get_transpose_b()) ?
please double check

Reply:
Yes, this version is correct. CPU FullyConnected operation is converted from the MatMul node with `transpose_a=false` and `transpose_b=true` so we create a separate transpose if `transpose_b=false`